lib/gssapi/krb5/name_attrs.c

```diff
@@ -0,0 +1,691 @@
+            urn.value = l;                                              \
+            urn.length = sizeof(l) - 1;                                 \
+        } else {                                                        \
+            urn.value = GSS_KRB5_NAME_ATTRIBUTE_BASE_URN l;             \
+            urn.length = sizeof(GSS_KRB5_NAME_ATTRIBUTE_BASE_URN l) - 1;\
+        }                                                               \
+        major = add_urn(minor_status, name, &urn, attrs);               \
+    } while (0)
+
+OM_uint32 GSSAPI_CALLCONV
+_gsskrb5_inquire_name(OM_uint32 *minor_status,
+                      gss_name_t name,
+                      int *name_is_MN,
+                      gss_OID *MN_mech,
+                      gss_buffer_set_t *attrs)
+{
+    OM_uint32 major = GSS_S_COMPLETE;
+    gss_buffer_desc urn;
+    krb5_error_code ret;
+    krb5_context context;
+    char lname[32];
+
+    GSSAPI_KRB5_INIT(&context);
+
+    *minor_status = 0;
+    if (name_is_MN)
+        *name_is_MN = 1;
+    if (MN_mech)
+        *MN_mech = GSS_KRB5_MECHANISM;
+    if (name == GSS_C_NO_NAME)
+        return GSS_S_CALL_INACCESSIBLE_READ;
+    if (attrs == NULL)
+        return GSS_S_CALL_INACCESSIBLE_WRITE;
+    ADD_URN("realm");
+    ADD_URN("name-ncomp");
+    ADD_URN("name-ncomp#0");
+    ADD_URN("name-ncomp#1");
+    ADD_URN("name-ncomp#2");
+    ADD_URN("name-ncomp#3");
+    ADD_URN("name-ncomp#4");
+    ADD_URN("name-ncomp#5");
+    ADD_URN("name-ncomp#6");
+    ADD_URN("name-ncomp#7");
+    ADD_URN("name-ncomp#8");
+    ADD_URN("name-ncomp#9");
+    ADD_URN("peer-realm");
+    ADD_URN("ticket-authz-data#pac");
+    ADD_URN("urn:mspac:");
+    ADD_URN("authenticator-authz-data"); /* XXX Add fragments? */
+    ADD_URN("ticket-authz-data"); /* XXX Add fragments? */
+    ADD_URN("authz-data");
+    ADD_URN("transit-path");
+    ADD_URN("canonical-name");
+    major = GSS_S_COMPLETE;
+    lname[0] = '\0';
+    ret = krb5_aname_to_localname(context, (void *)name,
+                                  sizeof(lname) - 1, lname);
+    if (ret == 0 && lname[0] != '\0')
+        major = gss_add_buffer_set_member(minor_status,
+                                          GSS_C_ATTR_LOCAL_LOGIN_USER, attrs); 
+    return major;
+}
+
+OM_uint32 GSSAPI_CALLCONV
+_gsskrb5_display_name_ext(OM_uint32 *minor_status,
+                          gss_name_t name,
+                          gss_OID display_as_name_type,
+                          gss_buffer_t display_name)
+{
+    krb5_const_principal p = (void *)name;
+    char *s = NULL;
+
+    *minor_status = 0;
+    if (display_name == NULL)
+        return GSS_S_CALL_INACCESSIBLE_WRITE;
+    display_name->length = 0;
+    display_name->value = NULL;
+
+    if (gss_oid_equal(display_as_name_type, GSS_C_NT_USER_NAME)) {
+        if (p->name.name_string.len != 1)
+            return GSS_S_UNAVAILABLE;
+        return _gsskrb5_localname(minor_status, name, GSS_KRB5_MECHANISM,
```

Comment:
`GSS_C_NT_USER_NAME` might be a local name in the standard, but in practice it's a synonym for `GSS_KRB5_NT_PRINCIPAL_NAME`. This wouldn't round-trip, not sure if I would advise this.

Reply:
Hmm, maybe, yeah.  Round-tripping is not a hard requirement.

Reply:
I'll keep this for now.  We're free to change this any time we want.